### PR TITLE
148820 bootstrap radix vulnerability scanner and api

### PR DESCRIPTION
**Pull request by nilsgstrabo**
Create required secrets, sql users and radix app for radix vulnerability scanner

### CODE REVIEW

File: scripts/install_base_components.sh [hunk 1]
```diff
@@ -73,6 +73,10 @@ hash htpasswd 2>/dev/null || {
   echo -e "\nERROR: htpasswd not found in PATH. Exiting..." >&2
   exit 1
 }
+hash sqlcmd 2>/dev/null || {
+  echo -e "\nError: sqlcmd not found in PATH. Exiting..."
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply scripts/install_base_components.sh:81]: fixed

File: scripts/migrate.sh [hunk 2]
```diff
@@ -66,6 +66,10 @@ hash flux 2>/dev/null || {
     echo -e "\nERROR: flux not found in PATH. Exiting... " >&2
     exit 1
 }
+hash sqlcmd 2>/dev/null || {
+    echo -e "\nError: sqlcmd not found in PATH. Exiting... " >&2
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply scripts/migrate.sh:70]: fixed

File: scripts/update_secret_for_vulnerability_scanner_api.sh [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+# INPUTS:
+#   RADIX_ZONE_ENV          (Mandatory)
+
+echo ""
+echo "Updating secret for the radix vulnerability scanner API"
+
+# Validate mandatory input
+
+if [[ -z "$RADIX_ZONE_ENV" ]]; then
+    echo "Please provide RADIX_ZONE_ENV" >&2
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply scripts/update_secret_for_vulnerability_scanner_api.sh:22]: fixed

File: scripts/update_secret_for_vulnerability_scanner_api.sh [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    echo "SQL_SERVER=$VULNERABILITY_SCAN_SQL_SERVER_FQDN
+    SQL_DATABASE=$VULNERABILITY_SCAN_SQL_DATABASE_NAME
+    SQL_USER=$VULNERABILITY_SCAN_SQL_API_USER
+    SQL_PASSWORD=$VULNERABILITY_API_SQL_LOGIN_PASSWORD
+    " > radix-vulnerability-scanner-api-secrets.env
+
+    VULNERABILITY_SCANNER_API_SECRET_NAME_QA=$(kubectl get secret --namespace "radix-vulnerability-scanner-api-qa" --selector radix-component="server" -ojson | jq -r .items[0].metadata.name)
+
+    if [[ -z "$VULNERABILITY_SCANNER_API_SECRET_NAME_QA" ]]; then
+        echo "ERROR: Could not get secret for server component in radix-vulnerability-scanner-api-qa."
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply scripts/update_secret_for_vulnerability_scanner_api.sh:63]: fixed

File: scripts/vulnerability-scanner/functions.sh [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        >&2 printf "${red}$(cat error.txt)\n"
+        rm -f error.txt
+        return 1
+    fi
+
+    rm -f error.txt 
+
+    local password=$(python3 -c 'import os,base64; print(base64.urlsafe_b64encode(os.urandom(32)).decode())') || exit
+
+    az keyvault secret set --vault-name $keyvault --name $secretName --value $password > /dev/null || exit
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply scripts/vulnerability-scanner/functions.sh:56]: fixed

File: scripts/vulnerability-scanner/bootstrap_scanner.sh [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+    || exit
+
+
+echo "Install Radix Vulnerability Scanner resources for flux"
+
+az keyvault secret download \
+    --vault-name "$AZ_RESOURCE_KEYVAULT" \
+    --name "${AZ_SYSTEM_USER_CONTAINER_REGISTRY_READER}" \
+    --file acr_sp_credentials.json \
+    || exit
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply scripts/vulnerability-scanner/bootstrap_scanner.sh:219]: fixed

File: scripts/vulnerability-scanner/functions.sh [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+    # az cli returns exit code 3 when secret does not exist.
+    # If exit code is not 3 then something else went wrong (e.g. auth or invalid secret name) and we stop the script
+    if [[ $status -gt 0 && $status -ne 3 ]]; then
+        >&2 printf "${red}$(cat error.txt)\n"
+        rm -f error.txt
+        return 1
+    fi
+
+    rm -f error.txt
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply scripts/vulnerability-scanner/functions.sh]: fixed

File: scripts/vulnerability-scanner/functions.sh [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+        return 1
+    fi
+
+    userName=$userName password=$password roles=$roles sqlcmd -b \
+        -S $serverName \
+        -d $databaseName \
+        -U $connectLoginName \
+        -P $connectPassword \
+        -i "create_or_update_user.sql" \
+        || exit
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply scripts/vulnerability-scanner/functions.sh:103]: fixed

File: scripts/vulnerability-scanner/functions.sh [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+
+    myip=$(curl http://ifconfig.me/ip) || exit
+
+    az sql server firewall-rule create \
+        --end-ip-address $myip \
+        --start-ip-address $myip \
+        --name ${ruleName} \
+        --resource-group $resourceGroup \
+        --server $server \
+        > /dev/null
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply scripts/vulnerability-scanner/functions.sh]: fixed

File: scripts/vulnerability-scanner/functions.sh [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+
+    az sql server firewall-rule create \
+        --end-ip-address $myip \
+        --start-ip-address $myip \
+        --name ${ruleName} \
+        --resource-group $resourceGroup \
+        --server $server \
+        --output none \
+        --only-show-errors ||
+        { echo"ERROR: Failed to create firewall rule $ruleName." >&2; return 1; }
```
Comment (oleterno-equinor) [inline scripts/vulnerability-scanner/functions.sh:121]: insert space between command and string
```suggestion
        { echo "ERROR: Failed to create firewall rule $ruleName." >&2; return 1; }
```

File: scripts/vulnerability-scanner/functions.sh [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+    server=$1
+    resourceGroup=$2
+    ruleName=$3
+
+    az sql server firewall-rule delete \
+        --name ${ruleName} \
+        --resource-group $resourceGroup \
+        --server $server \
+        --only-show-errors ||
+        { echo"ERROR: Failed to delete firewall rule $ruleName." >&2; return 1; }
```
Comment (oleterno-equinor) [inline scripts/vulnerability-scanner/functions.sh:134]:
```suggestion
        { echo "ERROR: Failed to delete firewall rule $ruleName." >&2; return 1; }
```

File: scripts/vulnerability-scanner/functions.sh [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+delete_sql_firewall_rule() {
+    server=$1
+    resourceGroup=$2
+    ruleName=$3
+
+    az sql server firewall-rule delete \
+        --name ${ruleName} \
+        --resource-group $resourceGroup \
+        --server $server \
+        --only-show-errors ||
```
Comment (oleterno-equinor) [inline scripts/vulnerability-scanner/functions.sh:133]: add `--output none`
```suggestion
        --output none
        --only-show-errors ||
```

**Review by oleterno-equinor — APPROVED** [review]

**Pull request merged**